### PR TITLE
Add Transifex push script

### PR DESCRIPTION
**Pull request by amangano-edx**
Adds script to extract translations strings and push them to Transifex.

Depends on https://github.com/edx/ecommerce-scripts/pull/60 and https://github.com/edx/ecommerce-scripts/pull/55

Replaces https://github.com/edx/ecommerce-scripts/pull/57

### CODE REVIEW

File: transifex/pull.py [hunk 1]
```diff
@@ -6,238 +6,55 @@
 To use, export an environment variable `GITHUB_ACCESS_TOKEN`. The token requires
 GitHub's "repo" scope.
 
-Run the script from the root of this repo. To use the repo clone URLs stored in
-settings.yaml, simply run:
+Run the script from the root of this repo.
 
-    python transifex/pull.py
+    python transifex/pull.py git@github.com:edx/course-discovery.git
```
Comment (BenjiLee) [inline transifex/pull.py]: Why is this hardcoded to the course-discovery repo?

**Comment by BenjiLee** [comment]
Will review #55 and #60 before this PR.

File: transifex/push.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+"""
+This script can be used to automatically extract translation strings,
+commit, push, and merge them to their respective repos, and then push them to transifex.
+
+To use, export an environment variable `GITHUB_ACCESS_TOKEN`. The token requires
+GitHub's "repo" scope.
+
+Run the script from the root of this repo.
+
+    python3 transifex/push.py git@github.com:edx/course-discovery.git
```
Comment (rlucioni) [inline transifex/push.py]: Following up on https://github.com/edx/ecommerce-scripts/pull/57#discussion_r123099995: you've already denoted the expected version of Python with the shebang line above. You'll typically see the expected version of Python aliased to `python` in most virtualenvs. Do whatever you're comfortable with here, but leaving it as `python` is sufficient.

File: transifex/push.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+If you want to use a custom merge method pass the --merge-method option.
+
+    python3 transifex/push.py git@github.com:edx/course-discovery.git --merge-method rebase
+"""
+from argparse import ArgumentParser
+from utils import DEFAULT_MERGE_METHOD, MERGE_METHODS, logger, repo_context
```
Comment (rlucioni) [inline transifex/push.py]: Minor, but please group your imports correctly. isort can help if you're not sure.

File: transifex/utils.py [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+                        self.name, self.pr.number, retries, self.MAX_MERGE_RETRIES
+                    )
+
+                    retries += 1
+
+                    if retries <= self.MAX_MERGE_RETRIES:
+                        # Poll every 5 minutes
+                        time.sleep(60 * 5)
+        else:
+            logger.info(
```
Comment (bderusha) [inline transifex/utils.py]: Should this be louder than just `info`?  If the retries have failed it's possible that github is down and the comment might also fail.

File: transifex/push.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+# The name of the branch to use.
+BRANCH_NAME = 'update-translation-strings'
+
+# The commit message to use.
+MESSAGE = 'Update translation strings'
+
+
+def push(clone_url, merge_method=None):
+    """Extracts translations for the given repo, commits them, pushes them to GitHub, opens a PR, waits for status
```
Comment (BenjiLee) [inline transifex/push.py]: When does this method merge and delete the PR?

File: transifex/utils.py (hunk 4 above)
Comment (amangano-edx) [inline transifex/utils.py]: I would prefer not to address this here, as this is one of several issues I noticed with the existing script that I decided not to do anything about during this pass of refactoring. I have created issues on the repository for each of the problems I identified that I would like to see fixed going forward. See https://github.com/edx/ecommerce-scripts/issues

File: transifex/push.py (hunk 5 above)
Comment (amangano-edx) [inline transifex/push.py]: Merge step: https://github.com/edx/ecommerce-scripts/blob/e9f5a87cca02b7f8d782f755945730b9a7413bb5/transifex/push.py#L37
Delete PR step: https://github.com/edx/ecommerce-scripts/blob/e9f5a87cca02b7f8d782f755945730b9a7413bb5/transifex/utils.py#L69

**Pull request merged**